viostor/virtio_pci.c

```diff
@@ -197,12 +197,12 @@ static u16 vdev_get_msix_vector(void *context, int queue)
             } else {
                 vector = queue + 1;
             }
-        } else {
-            vector = VIRTIO_MSI_NO_VECTOR;
         }
     } else {
         /* on-device-config-change interrupt */
-        vector = VIRTIO_MSI_NO_VECTOR;
+        if (!adaptExt->msix_one_vector) {
```

Comment:
Do you think that failing driver to load (especially when it provides 99.99% of its functionality but fails to handle configuration changed request which probably never happens in its life) will be a good solution?

In the real life, failing to report the correct number of vectors  vector is a HW firmware bug, which makes a driver to deal with a broken hardware. Keeping a driver functional in this situation is a big win, but the HW has to be fixed ASAP. Do you have any particular scenario in your mind when a user will prefer using one vector, rater that specifying number of vectors equal to number of queues + 1, or switching to IRQ mode?

As I  asked already, do you really think that failing a boot-time driver because it will not be able to handle one minor feature is what we should to do?  Regarding to looking at source of interrupt, you probably mean reading the ISR status register. Am I right? If so, who can guarantee that ISR status register will provide us with the correct information now and in the future? And even more, IO operations in MSI service routine looks like a bit strange solution to me.